spec/services/invoices/payments/adyen_service_spec.rb

```diff
@@ -0,0 +1,244 @@
+
+      it 'does not creates a adyen payment' do
+        result = adyen_service.create
+
+        expect(result).to be_success
+
+        aggregate_failures do
+          expect(result.invoice).to eq(invoice)
+          expect(result.payment).to be_nil
+
+          expect(payments_api).not_to have_received(:payments)
+        end
+      end
+    end
+
+    context 'with error on adyen' do
+      let(:customer) { create(:customer, organization:) }
+
+      let(:subscription) do
+        create(:subscription, organization:, customer:)
+      end
+
+      let(:organization) do
+        create(:organization, webhook_url: 'https://webhook.com')
+      end
+
+      before do
+        subscription
+
+        allow(payments_api).to receive(:payments)
+          .and_raise(Adyen::AdyenError.new(nil, nil, 'error', 'code'))
+      end
+
+      it 'delivers an error webhook' do
+        expect { adyen_service.send(:create_adyen_payment) }
+          .to raise_error(Adyen::AdyenError)
+
+        expect(SendWebhookJob).to have_been_enqueued
+          .with(
+            'invoice.payment_failure',
+            invoice,
+            provider_customer_id: adyen_customer.provider_customer_id,
+            provider_error: {
+              message: 'error',
+              error_code: 'code',
+            },
+          )
+      end
+    end
+  end
+
+  describe '.update_payment_status' do
+    let(:payment) do
+      create(
+        :payment,
+        invoice:,
+        provider_payment_id: 'ch_123456',
+        status: 'Pending',
+      )
+    end
+
+    before do
+      allow(SendWebhookJob).to receive(:perform_later)
+      payment
+    end
+
+    it 'updates the payment and invoice payment_status' do
+      result = adyen_service.update_payment_status(
+        provider_payment_id: 'ch_123456',
+        status: 'Authorised',
+      )
+
+      expect(result).to be_success
+      expect(result.payment.status).to eq('Authorised')
+      expect(result.invoice.reload).to have_attributes(
+        payment_status: 'succeeded',
+        ready_for_payment_processing: false,
+      )
+    end
+
+    context 'when status is failed' do
+      it 'updates the payment and invoice status' do
+        result = adyen_service.update_payment_status(
+          provider_payment_id: 'ch_123456',
+          status: 'Refused',
+        )
+
+        expect(result).to be_success
+        expect(result.payment.status).to eq('Refused')
+        expect(result.invoice.reload).to have_attributes(
+          payment_status: 'failed',
+          ready_for_payment_processing: true,
+        )
+      end
+    end
+
+    context 'when invoice is already succeeded' do
+      before { invoice.succeeded! }
+
+      it 'does not update the status of invoice and payment' do
+        result = adyen_service.update_payment_status(
+          provider_payment_id: 'ch_123456',
+          status: ['Authorised', 'SentForSettle', 'SettleScheduled', 'Settled', 'Refunded'].sample
```

Comment:
Style/WordArray: Use `%w` or `%W` for an array of words. (https://rubystyle.guide#percent-w)